cmd/loop/main.go

```diff
@@ -374,6 +392,9 @@ func getClientConn(address, tlsCertPath, macaroonPath string) (*grpc.ClientConn,
 
 	opts := []grpc.DialOption{
 		grpc.WithDefaultCallOptions(maxMsgRecvSize),
+
+		// We always need to send a macaroon.
+		readMacaroon(macaroonPath),
```

Comment:
`getClientConn` has an error return value, so you could pass back an error from `readMacaroon`